azure-proto-core/DeletableResourceOperations.cs

```diff
@@ -0,0 +1,17 @@
+using Azure;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace azure_proto_core
+{
+    public interface IDeletableResource<TOperations, TResource> 
+        where TResource:Resource 
+        where TOperations: IDeletableResource<TOperations, TResource>
+    {
+        ArmOperation<Response> Delete();
```

Comment:
Should we have StartDelete, StartDeleteAsync, Delete, DeleteAsync?

Reply:
Doesn't Startxxx only apply to certain RPs that do LROs?

Reply:
I think we should do this a seperate task since, VirtualNetworkOperations, NetworkInterfaceOperations, ArmOperation,  and VirtualMachineOperations all need update to implement this interface when we add StartDelete.

Reply:
https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_sprints/backlog/DotNET%20Management%20SDK%20Team/DotNET%20Management%20SDK/FE/180?workitem=4883